ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1753,43 +1761,61 @@ private class JdbcLedgerDao(
 
 object JdbcLedgerDao {
 
-  val defaultNumberOfShortLivedConnections = 16
+  private val DefaultNumberOfShortLivedConnections = 16
+
+  def readOwner(
+      jdbcUrl: String,
+      metrics: MetricRegistry,
+  )(implicit logCtx: LoggingContext): ResourceOwner[LedgerReadDao] = {
+    val maxConnections = DefaultNumberOfShortLivedConnections
+    owner(jdbcUrl, maxConnections, metrics)
+      .map(new MeteredLedgerReadDao(_, metrics))
+  }
 
-  def owner(
+  def writeOwner(
       jdbcUrl: String,
       metrics: MetricRegistry,
-      executionContext: ExecutionContext,
   )(implicit logCtx: LoggingContext): ResourceOwner[LedgerDao] = {
     val dbType = DbType.jdbcType(jdbcUrl)
     val maxConnections =
-      if (dbType.supportsParallelWrites) defaultNumberOfShortLivedConnections else 1
-    for {
-      dbDispatcher <- DbDispatcher.owner(jdbcUrl, maxConnections, metrics)
-    } yield new MeteredLedgerDao(JdbcLedgerDao(dbDispatcher, dbType, executionContext), metrics)
+      if (dbType.supportsParallelWrites) DefaultNumberOfShortLivedConnections else 1
+    owner(jdbcUrl, maxConnections, metrics)
+      .map(new MeteredLedgerDao(_, metrics))
   }
 
-  def apply(
-      dbDispatcher: DbDispatcher,
-      dbType: DbType,
-      executionContext: ExecutionContext,
-  )(implicit logCtx: LoggingContext): LedgerDao =
-    new JdbcLedgerDao(
-      dbDispatcher,
-      ContractSerializer,
-      TransactionSerializer,
-      KeyHasher,
-      dbType,
-      executionContext,
-    )
+  private def owner(
+      jdbcUrl: String,
+      maxConnections: Int,
+      metrics: MetricRegistry,
+  )(implicit logCtx: LoggingContext): ResourceOwner[LedgerDao] =
+    for {
+      dbDispatcher <- DbDispatcher.owner(jdbcUrl, maxConnections, metrics)
+      executor <- ResourceOwner.forExecutorService(
+        () =>
+          Executors.newCachedThreadPool(
+            new ThreadFactoryBuilder().setNameFormat("dao-executor-%d").build()))
```

Comment:
I think it would be better to make this a `private def` at the top level, with a name so that it's clear where we're building this.

Reply:
Done. Made it a `val` because thread factories are stateless.